test/integration/deployment/deployment_test.go

```diff
@@ -381,3 +381,70 @@ func TestDeploymentHashCollision(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestOverlappingDeployments(t *testing.T) {
+	s, closeFn, rm, dc, informers, c := dcSetup(t)
+	defer closeFn()
+	name := "test-overlapping-deployments"
+	ns := framework.CreateTestingNamespace(name, s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	replicas := int32(1)
+	firstDeploymentName := "first-deployment"
+	secondDeploymentName := "second-deployment"
+	testers := []*deploymentTester{
+		{t: t, c: c, deployment: newDeployment(firstDeploymentName, ns.Name, replicas)},
+		{t: t, c: c, deployment: newDeployment(secondDeploymentName, ns.Name, replicas)},
+	}
+	testers[1].deployment.Spec.Template.Labels["other-label"] = "random-label"
```

Comment:
This can be removed, because 2 deployments already have overlapping selectors before this line (one's selector matches the other one's pod labels).

Reply:
The `other-label` was there for original e2e test. I thought it was to differentiate the two deployments (they have similar but not-exactly-same selector labels). I will remove it.